simple_salesforce/api.py

```diff
@@ -207,6 +212,20 @@ def __init__(
 
         self.api_usage = {}
 
+    def _generate_headers(self):
+        self.headers = {
+            'Content-Type': 'application/json',
+            'Authorization': 'Bearer ' + self.session_id,
+            'X-PrettyPrint': '1'
+        }
+
+    def _refresh_session(self):
```

Comment:
Run the login logic cached in `__init__` and recalculate the headers with the new `session_id`.